src/main/java/com/synopsys/integration/jira/common/rest/service/JiraApiClient.java

```diff
@@ -184,7 +185,12 @@ private JiraRequest createDeleteRequest(HttpUrl url) {
 
     private <R extends IntRestComponent> R parseResponse(JiraResponse jiraResponse, Type type) throws IntegrationException {
         String content = jiraResponse.getContent();
-        return jsonTransformer.getComponentAs(content, type);
+        R parsedResponse = jsonTransformer.getComponentAs(content, type);
+        if (parsedResponse instanceof IntRestResponse) {
+            // Necessary because we don't call jsonTransformer.getResponse(...)
+            ((IntRestResponse) parsedResponse).setGson(gson);
```

Comment:
This seems odd. Is this because we don't know if the JiraResponse represents a IntRestResponse or a IntRestComponent?

Reply:
To answer your question: yes, but I'm not sure why; regardless, it's out of scope for this PR to refactor that structure.